app/controllers/submissions_controller.rb

```diff
@@ -469,7 +469,7 @@ def get_feedback_file
   end
 
   def download
-    @assignment = Assignment.find(params[:id])
+    @assignment = Assignment.find(params[:assignment_id])
```

Comment:
DId you double-check to make sure that this change doesn't affect other uses of the route? E.g. downloading the file by clicking on the filename in the student submissions view.

Reply:
On line 95 of submissions_helper.rb the url for download is passed back from `get_file_info`, which is indirectly used by `#populate_file_manager` and now I am thinking if there is a reason we are making the url in the fileviewer component at all?

Reply:
Hmm, isn't the `url` state for `FileViewer` the URL that's passed to `ImageViewer` and `PDFViewer`? Or is that a different URL?